### DISCUSSION TITLE
.NET 8, Cannot deploy on IOS Simulator - nothing happens

### DISCUSSION THREAD

**Discussion by SergTomcat** [dotnet/maui#19215] [Q&A] [answered]
After upgrading to .net 8 (8.0.4) and Visual Studio 2022 17.8.2, app is building and saying that it is deployed, but nothing happens on connected Mac side, simulator not launching or receive app.

VS output window:
12:17:38:589	========== Build: 1 succeeded, 0 failed, 2 up-to-date, 0 skipped ==========
12:17:38:589	========== Build completed at 12:17 and took 29,930 seconds ==========
12:17:38:590	========== Deploy: 1 succeeded, 0 failed, 0 skipped ==========
12:17:38:590	========== Deploy completed at 12:17 and took 29,930 seconds ==========

**Comment by mobilewares** [comment]
Same thing is broken here since .NET8 upgrade for our project.  You really need to compile/deploy from VS Mac. 

If you don't need to debug App from Windows while running and you want it to run in the remote simulator (so you can see it on Windows PC) - create a new MAUI App - do remote launch so iOS Remoted simulator comes up on Windows (for some reason this works). Then on the Mac run/deploy it to same simulator & you will be able to control it and use it from Windows - you can even debug it from VS Mac if you want but that's a pretty crappy IDE.

**Comment by SergTomcat** [comment, marked as answer]
Found out that in old projects (before .NET 8) there is global.json file with mentioned framework version. Change version or, better, delete it completely and it will begin deploying fine.

**Reply by tklempay** [reply]
just updated a Maui.iOS (formerly Xamarin iOS Native) app from .NET 6 to .NET 8.  Had the same issue, applied your fix and simulator started right up.  Thank you!

**Reply by mglass1** [reply]
Hello, I am experiencing this same issue but am not able to locate the global.json file. Where did you find this file?

**Reply by tklempay** [reply]
you can create it by hand

**Reply by mglass1** [reply]
Thank you for the quick response. Where do you create this file and do you need to add any content?

**Comment by archergod** [comment]
I have similar issue with 17.8.5 and there is not Global.json file. Is there any other possible solution to this ?

**Reply by archergod** [reply]
I again have this issue with 17.9.6 .. Why it keep coming back.

**Comment by YARG** [comment]
I'd been running against the iOS simulator for months with no problem, I then did a Release Build for testing, which meant that I had an "iPhone Distribution" provisioning profile selected under the "Signing Identity" option (under Properties -> iOS -> Bundle Signing).  

I then tried to run against the Simulator in Debug and VS2022 (Version 17.9.6) 'said' that the deployment to the simulator had been successful, which it hadn't .. nothing fired up.   As a test I deleted the obj/bin, tried again, same thing.  Switched the "Signing Identity" back to "Developer (Automatic) and provisioning profile to Automatic and the Simulator fired up again.   I hope this helps.

![Screenshot 2024-06-11 at 17 01 05](https://github.com/dotnet/maui/assets/568435/e5e92a46-ff75-49be-be00-c15cca035d8d)

**Reply by archergod** [reply]
I have similar issue, I use direct `csproj` file editing and manually set those value. Plus then I use VS code on mac itself to publish release or debug. Visual Studio seems all broken and generate random issue with cache and stuff. I get the Simulator runs once per Reboot of both my mac machine and Windows machine. but it works charm on VS Code on Mac itself.

**Reply by janvans** [reply]
I had the same issue - after doing a deploy to TestFlight I was not able to run the app in debug mode on a simulator again - everything seemed fine and it was deployed "successfully", but the simulator did not open, and even after opening the simulator manually, the app was not installed there. Although I've switched back from Manual to Automatic provisioning, I found the following PropertyGroup in the project file (Edit Project File) :

< PropertyGroup Condition="'$(TargetFramework)'=='net8.0-ios'">
     < ProvisioningType>manual< /ProvisioningType>
     < CodesignKey>iPhone Distribution: XXXXXXX)< /CodesignKey>
     < CodesignProvision>YYYYYYY< /CodesignProvision>
< /PropertyGroup>

I deleted this whole code block, saved and re-selected Automatic Provisioning using the visual project file editor, and everything worked as expected.
(Visual Studio v17.10.3 XCode v15.2)

**Reply by archergod** [reply]
After spending week on the issue, I can confirm that Provisioning setting are not solely responsible for it. I have those set and it works. The problem is majorly because of cache. Delete bin/obj folder in main project and all sub projects, and then if possible find and delete from your mac machine. Reboot both Windows and or mac machine and then probably it works. 

I create a new out of box maui project match and remove anything extra from my project file to match it and then I repeat those clear cache steps. And it start to compile. 

Also if you stop debugger in between or it get disconnected or crash because of any code error. Then chances are it won't compile again properly. Now I delete 5-8 times Cached Bin/obj folder a day to get it working (which make my recycle bin like 10 GB by end of day)

**Comment by francoZuniga32** [comment]
I'm having exactly the same problem, I don't get any errors when compiling but the simulator doesn't launch.
I tried deleting the obj and bin folders, and it doesn't work either.
I have automatic provisioning configured in the project configuration and it doesn't work either.
I also saw that some lines are generated in the .cproj and it wouldn't work either.

![github-cap](https://github.com/user-attachments/assets/224fe23c-139b-46a9-92eb-cb12c6b36ab5)
![github-cap-2](https://github.com/user-attachments/assets/b5b5f37a-6f38-49a8-9bb9-cc40dce7c283)

**Comment by ColinBriggs** [comment]
Same symptoms. Compiles OK, no errors. But never starts the iOS simulator. Removing the Automatic Provisioning and deleting the Bin and Obj folders did not work. The problem in my case was the wrong certificate in the Keychain on the MAC. I removed the wrong one and installed the correct one. Then it all worked OK.

**Comment by stephenquan** [comment]
N.B. Visual Studio for Mac has been retired as of August 31, 2024

The approach that we follow for iOS development, particularly for new developers being setup on new macs is:
 - build and deploy a sample native iOS app in Xcode to iOS Simulator and resolve any Xcode and/or iOS developer certificate issue
 - install Visual Studio Code and the .NET Meteor extension (.NET MAUI extension should also work)
 - build and deploy a sample .NET MAUI application to the same iOS Simulator that you did with Xcode
 - finally your project should work in either Visual Studio Code and/or from Visual Studio with pair-to-mac
